### ISSUE TITLE
Interrupting drivers breaks cluster

### ISSUE THREAD

**Issue by ericl** [ray-project/ray#1555]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **Ray installed from (source or binary)**: source
- **Ray version**: master
- **Python version**: 2.7
### Describe the problem

1. Setup a cluster
2. Run some script, e.g. ./train.py -f tuned_examples/pong_ppo.yaml --redis-address=<>
3. Interrupt the script (e.g. ctrl-C or ctrl-\ if it doesn't work)
4. Try this a few times. Eventually either (1) you'll get a resource error that the cluster has insufficient GPUs or something, or (2) 
```
This worker or driver is waiting to receive a class definition so that it can deserialize an object from the object store. This may be fine, or it may be a bug.

  You can inspect errors by running

      ray.error_info()

  If this driver is hanging, start a new one with

      ray.init(redis_address="172.30.0.170:6379")
```

At this point you have to `ray create_or_update` your cluster again.

**Comment by virtualluke** [comment]
I see this also with python 3.6 on redhat 7.4

**Comment by robertnishihara** [comment]
This may be fixed by #1927.

**Comment by robertnishihara** [comment]
Fixed I think.
